temporal-sdk/src/main/java/io/temporal/common/interceptors/WorkflowClientInterceptor.java

```diff
@@ -23,15 +23,47 @@
 import io.temporal.client.ActivityCompletionClient;
 import io.temporal.client.WorkflowOptions;
 import io.temporal.client.WorkflowStub;
+import io.temporal.common.Experimental;
 import java.util.Optional;
 
+@Experimental
 public interface WorkflowClientInterceptor {
-
+  /**
+   * Called when workflow stub is instantiated during creation of new workflow. It allows to
+   * decorate calls to {@link WorkflowStub} instance which is an entry point for client code.
+   *
+   * @return decorated stub
+   * @deprecated consider implementing all intercepting functionality using {@link
+   *     WorkflowClientCallsInterceptor} that is produced in {@link
+   *     #workflowClientClassInterceptor}. This method has to stay temporary because
+   *     TimeLockingInterceptor has to intercept top level {@link WorkflowStub} methods.
+   */
+  @Deprecated
```

Comment:
Should we mark this `forRemoval`?

Reply:
We could, but it looks like your project java language level is 1.8
forRemoval is java 9

Reply:
Ah, that's a bummer.